Remotion/Web/Development.WebTesting/WebTestSetUpFixtureHelper.cs

```diff
@@ -144,42 +144,46 @@ private void VerifyWebApplicationStarted (Uri webApplicationRoot, TimeSpan appli
 
       var stopwatch = Stopwatch.StartNew();
 
-#pragma warning disable SYSLIB0014
-      var webRequest = (HttpWebRequest)HttpWebRequest.Create(resolvedUri); // TODO RM-8492: Replace with HttpClient
-#pragma warning restore SYSLIB0014
-      webRequest.Method = WebRequestMethods.Http.Head;
-      webRequest.AllowAutoRedirect = true;
-      webRequest.Host = webApplicationRoot.Host;
-      webRequest.ServerCertificateValidationCallback += (sender, certificate, chain, errors) => true;
-
       HttpStatusCode statusCode = default;
       Assertion.DebugAssert(statusCode != HttpStatusCode.OK);
 
-      while (statusCode != HttpStatusCode.OK)
+      while (true)
       {
+#pragma warning disable SYSLIB0014
+        var webRequest = (HttpWebRequest)HttpWebRequest.Create(resolvedUri); // TODO RM-8492: Replace with HttpClient
+#pragma warning restore SYSLIB0014
+        webRequest.Method = WebRequestMethods.Http.Head;
+        webRequest.AllowAutoRedirect = true;
+        webRequest.Host = webApplicationRoot.GetComponents(UriComponents.Host | UriComponents.Port, UriFormat.UriEscaped);
+        webRequest.ServerCertificateValidationCallback += (sender, certificate, chain, errors) => true;
+
+        var remainingTimeout = (int)(applicationPingTimeout.TotalMilliseconds - stopwatch.Elapsed.TotalMilliseconds);
+        if (remainingTimeout <= 0)
+        {
+          throw new WebException(
```

Comment:
I think we should add a logstatement here, since we have one in the exception-case, too.